tests/testthat/test-uncount.R

```diff
@@ -22,7 +22,33 @@ test_that("expands constants and expressions", {
   expect_equal(uncount(df, 1 + 1), df[c(1, 1), ])
 })
 
-test_that("must evaluate to numeric", {
+
+test_that("works with groups", {
+  df <- tibble(g = 1, x = 1, w = 1) %>% dplyr::group_by(g)
+  expect_equal(uncount(df, w), df %>% dplyr::select(-w))
+})
+
+test_that("doesn't remove grouping variable", {
+  df <- tibble(g = 1, x = 1, w = 1) %>% dplyr::group_by(g)
+
+  expect_message(uncount(df, g)) %>%
+    expect_equal(df)
+})
+
+test_that("must evaluate to integer", {
+  df <- tibble(x = 1, w = 1/2)
+  expect_error(uncount(df, w), class = "vctrs_error_cast_lossy")
+
   df <- tibble(x = 1)
-  expect_error(uncount(df, "W"), "must evaluate to a numeric vector")
+  expect_error(uncount(df, "W"), class = "vctrs_error_incompatible_type")
+})
+
+test_that("works with 0 weights", {
+  df <- tibble(x = 1:2, w = c(0, 1))
+  expect_equal(uncount(df, w), tibble(x = 2))
+})
+
+test_that("errors on negative weights", {
```

Comment:
`vec_rep_each()` will already error on negative `times`. We should probably consider adding `x_arg` and `times_arg` arguments to `vec_rep_each()` so the message can be improved and used here. What you have now is probably fine in the meantime.

```r
> vec_rep_each(df, df$w)
Error: `times` must be a vector of positive numbers. Location 1 is negative.
```

Same with `NA` times

```r
> vec_rep_each(df, NA)
Error: `times` can't be missing. Location 1 is missing.
```

Reply:
Yes, that would be nice. I added a reminder for that in the code.